Search, One-page forms and User mgmt section review (#10)
ezsystems/commerce-documentation: docs/guide/search/search_features/search_indexing/indexing_file_content.md

# Indexing file content

Indexing the content of files stored in the content model uses an additional component called Apache Tika.

> Apache Tika has two available modes: Server Mode and App Mode. eZ Commerce are using Apache Tika in App mode.

Reviewer: ```suggestion
Apache Tika has two available modes: Server Mode and App Mode. eZ Commerce is using Apache Tika in App mode.
```

> Apache Tika has two available modes: Server Mode and App Mode. eZ Commerce are using Apache Tika in App mode.

Author: ```suggestion
Apache Tika has two available modes: Server Mode and App Mode. eZ Commerce uses Apache Tika in App mode.
```

You can specify which MIME types are indexed in configuration:

``` yaml
siso_search.default.index_content:
    - application/pdf
```

If you need additional file types to be indexed, add them to this configuration, for example:

``` yaml
siso_search.default.index_content:
    - application/pdf
    - application/vnd.ms-excel
    - application/msword
```

See [Supported Document Formats](http://tika.apache.org/1.13/formats.html) for a list of formats supported by Apache Tika.

!!! caution

    Make sure [Apache Tika Bundle](https://packagist.org/packages/jolicode/apache-tika-bundle) is installed

>     and its bundle is enabled in the kernel.

Reviewer: ```suggestion
    and is enabled in the kernel.
```

    Apache Tika log file (`/tmp/tika-error.log`) is hardcoded. Make sure the file is writable both by the command-line user who executes the indexer

>     and the Apache user, because the indexer is triggered after a file is modified modification in the backend.

Reviewer: ```suggestion
    and the Apache user, because the indexer is triggered after a file is modified in the backend.
```
